pkg/repositories/repositories.go

```diff
@@ -58,6 +58,41 @@ func NewConfig(filename string) (Config, error) {
 	return repoConfig, nil
 }
 
+// UpdateConfigVersions updates a config in-place, setting the afterVersion to
+// its version, and it's version to a newer one specified in a secondary config
+func UpdateConfigVersions(baseConfig *Config, configWithNewVersions *Config) {
+	// Extract new repos regardless of category
+	var newRepos []Repository
+	for _, category := range configWithNewVersions.Section.Categories {
+		newRepos = append(newRepos, category.Repos...)
+	}
+
+	// Iterate through current categories and repositories, maintaining
+	// the structure.
+	for categoryIdx := range baseConfig.Section.Categories {
```

Comment:
I think this can be simplified, unless I'm missing something:
```
	for _, category := range baseConfig.Section.Categories {
		for repoIdx, currentRepo := range category.Repos {
			if newVersion := checkForNewVersion(newRepos, currentRepo); newVersion != "" {
				remappedRepo := currentRepo
				remappedRepo.AfterVersion = currentRepo.Version
				remappedRepo.Version = newVersion

				category.Repos[repoIdx] = remappedRepo
			}
		}
	}
}
```